app/controllers/public/sessions.js

```diff
@@ -17,24 +17,24 @@ export default class SessionsController extends Controller {
   @computed('model.session.@each', 'timezone')
   get groupByDateSessions() {
     let sessions;
-
     if (this.sort === 'title') {
       sessions = groupBy(this.model.session.toArray(), '');
     } else {
       sessions = groupBy(this.model.session.toArray(), s => moment.tz(s.startsAt, this.timezone).format('dddd, Do MMMM'));
     }
-
     return Object.entries(sessions).map(([date, sessions]) => ({ date: date === 'undefined' ? null : date, sessions }));
   }
 
-  @computed('model.event.startsAt', 'model.event.endsAt', 'timezone')
+  @computed('model.dates', 'timezone')
   get allDates() {
-    const arr = [];
-    const difference = (this.model.event.endsAt).diff(this.model.event.startsAt, 'days');
-    for (let i = 0; i <= Math.abs(difference); i++) {
-      arr.push(moment.tz(this.model.event.startsAt, this.timezone).add(i, 'days').toISOString());
-    }
-    return arr;
+
+    const uniqueDates = new Set();
+
+    this.model.dates.toArray().map(el => moment.tz(el.startsAt, this.timezone).toISOString()).sort().forEach(el => {
```

Comment:
No need of forEach, use .map

Reply:
i will use the .map here , but i am not getting , what's the difference , because .map will return the new array but forEach will only give the access to each element , i want to know about it?